test/cookie.test.js

```diff
@@ -847,3 +847,59 @@ test('should not decorate fastify, request and reply if no secret was provided',
     message: 'req.unsignCookie is not a function'
   })
 })
+
+test('disable parses incoming cookies', (t) => {
+  t.plan(6)
+  const fastify = Fastify()
+  fastify.register(plugin, { hook: false })
+
+  for (const hook of ['preValidation', 'preHandler']) {
+    fastify.addHook(hook, (req, reply, done) => {
+      t.equal(req.cookies, null)
+      done()
+    })
+  }
+
+  fastify.addHook('preParsing', (req, reply, payload, done) => {
+    t.equal(req.cookies, null)
+    done()
+  })
+
+  fastify.get('/disable', (req, reply) => {
+    t.equal(req.cookies, null)
+    reply.send()
+  })
+
+  fastify.inject({
+    method: 'GET',
+    url: '/disable',
+    headers: {
+      cookie: 'bar=bar'
+    }
+  }, (err, res) => {
+    t.error(err)
+    t.equal(res.statusCode, 200)
+  })
+})
+
+test('error if option is invalid', (t) => {
+  t.plan(1)
+  const fastify = Fastify()
+
+  fastify.register(plugin, { hook: true })
```

Comment:
I would have expected to throw already here. strange

Reply:
Also I had a typo in my suggestion. Should be "invalid" and not "invaid"

The whole test is actually this:

```typescript
test('result in an error if hook-option is set to an invalid value', async (t) => {
  t.plan(1)
  const fastify = Fastify()

  t.rejects(() => fastify.register(plugin, { hook: true }), new Error("You can set false or use only 'onRequest' , 'preParsing' , 'preValidation' , 'preHandler' hooks"))
})

```

Reply:
yes, it would indeed be better. corrected